PluginsAndFeatures/azure-toolkit-for-eclipse/com.microsoft.azuretools.core/src/com/microsoft/azuretools/core/handlers/SignInCommandHandler.java

```diff
@@ -37,11 +37,17 @@ public static boolean doSignIn(Shell shell) {
                 authMethodManager.setAuthMethodDetails(authMethodDetailsUpdated);
                 SelectSubsriptionsCommandHandler.onSelectSubscriptions(shell);
                 authMethodManager.notifySignInEventListener();
+                persistAuthMethodDetails();
             }
             return authMethodManager.isSignedIn();
         } catch (Exception e) {
             e.printStackTrace();
         }
         return false;
     }
+    
+    private static void persistAuthMethodDetails() {
+        AuthMethodManager.getInstance().persistAuthMethodDetails();
```

Comment:
Move the two method to `doSignIn`? IMO notify sign in listener should not be a part of persist authentication